### PR TITLE
rust, libnm_dbus: Make some improvements the libnm_dbus library

### PR DESCRIPTION
**Pull request by almusil**
- Use serde by proxy going through TryFrom when we are getting answer from dbus.  
- Remove some clones that are not needed
- Add macro for more convenient way to extract values from dbus `HashMap` responses

### CODE REVIEW

**Comment by almusil** [comment]
@ffmancera @cathay4t  
Please take a look

File: rust/src/libnm_dbus/connection/bridge.rs [hunk 1]
```diff
@@ -58,19 +58,20 @@ impl NmSettingBridge {
...
+#[serde(try_from = "DbusDictionary")]
 pub struct NmSettingBridgePort {
     _other: HashMap<String, zvariant::OwnedValue>,
 }
 
-impl TryFrom<HashMap<String, zvariant::OwnedValue>> for NmSettingBridgePort {
+impl TryFrom<DbusDictionary> for NmSettingBridgePort {
     type Error = NmError;
     fn try_from(
         setting_value: HashMap<String, zvariant::OwnedValue>,
```
Comment (cathay4t) [inline rust/src/libnm_dbus/connection/bridge.rs]: The TryFrom is using `DbusDictionary`, so the `fn try_from` should be also using `DbusDictionary`

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: rust/src/libnm_dbus/connection/bridge.rs (hunk 1 above)
Comment (almusil) [reply rust/src/libnm_dbus/connection/bridge.rs]: Good catch, thanks, done

**Review by cathay4t — APPROVED** [review]
LGTM.

**Pull request merged**